src/stream/mod.rs

```diff
@@ -469,45 +335,44 @@ impl StreamBuilder {
     pub fn start(self, token: &Token) -> TwitterStream {
         let mut params = HashMap::new();
 
-        if let Some(with_follows) = self.with_follows {
-            if with_follows {
-                add_param(&mut params, "with", "followings");
-            } else {
-                add_param(&mut params, "with", "user");
-            }
+        if let Some(filter_level) = self.filter_level {
+            add_param(&mut params, "filter_level", filter_level.to_string());
         }
 
-        if self.all_replies {
-            add_param(&mut params, "replies", "all");
+        if !self.follow.is_empty() {
+            let to_follow = self
+                .follow
+                .iter()
+                .map(|id| id.to_string())
```

Comment:
Since `self` is being consumed here, you can call `into_iter` and consume both this and `self.track` as well. That will save the memory allocation of cloning the Strings on `self.track`, though it won't help much here. (If you trade the `BTreeSet`s for plain `Vec`s, you can even save the reallocation on `self.track` entirely, and skip straight to calling `join`! Though you won't get the deduplication/sorting that the `BTreeSet` gives you, and i don't know how necessary that is for interacting with Twitter.)

Reply:
Agree, have switched to `Vec`. I thought it might be nice to have deduplication and ordering, but actually the user can deal with that, if they care (the twitter api doesn't).